### DISCUSSION TITLE
Preference for new task positioning (i.e. open next to the currently active one)

### DISCUSSION THREAD

**Discussion by Romasato** [lawand-dot-io/taskbar#71] [Feature requests]
Hello again,

Thank you for implementing the suggested changes in https://github.com/lawand-dot-io/taskbar/discussions/44 - that was very useful!

Now I have another suggestion in terms of the ordering of new tasks - hopefully something possible to implement.

At the moment a new Task is always added to the very end of Taskbar (right-most) after all the tasks.
It would be nice if we could have a configuration setting to control where newly opened Tasks appear:

- Default: Append to the end of taskbar.
- Alternative: Append right after currently active task.

This is useful when for example I am focused on Terminal app and open a new Terminal window for the same project dir - it would make sense for the new window to appear next to my already open Terminal window.

The way I order windows is grouping IDE+Terminal Tasks in twos, i.e. 
```
IDE (Project A) -> Terminal (Project A - one or more windows) -> IDE (project B) -> Terminal (Project B) -> ....
```
And quite often I open new Terminal window via IDE - but the new Terminal window gets appended to the end of Taskbar and I need to then manually re-order it next to IDE task.

What do you think? Is it feasible?

Thanks!

**Comment by lawand-dot-io** [comment]
Interesting idea. As always I will be implementing features that receive the highest number of upvotes first. Also, I am considering setting up a method for users to purchase new features such as this one in the future. This will allow me to implement basically all requested features. Will let you know if I went in that direction. Cheers

**Reply by Romasato** [reply]
Not sure how many actual users linger around here to vote on the new discussion threads, so it is best for yourself to decide what changes are the most important.
Also ultimately it is up to you how you want to monetize your app - and then users will decided what it is worth to them. Not sure how "purchasing" the new features development could work, but there are ways, i.e. crowdfunding. 

Speaking about my request specifically, I see it more as a UX niggle that I would like to see fixed than a new feature - a configuration option to retain previous functionality is how I would approach it just to maintain the existing behavior for users that got used to the initial behavior of new tasks.

Cheers :)

**Reply by Romasato** [reply]
Hi @lawand-dot-io,

Do you know if this one is being considered for implementation?

Thanks

**Reply by lawand-dot-io** [reply]
@Romasato hmm. I think this might be not be too hard to implement. I can simply do a programatic drag and drop of the new task after it's created, so that it's added right after the active one. I'll look into it and if this approach didn't cause technical complications I'll add it in the next version which should be out in a couple of months

**Reply by Romasato** [reply]
Hi @lawand-dot-io, 
not sure if the feature got added and I missed it somehow? Is it planned for one of the future versions possibly?

Thank you

**Reply by lawand-dot-io** [reply]
@Romasato thanks for the reminder, I'm not sure how I missed this one. Unfortunately at the moment taskbar's development is slowed down due to a personal health issue (https://x.com/lawand_dot_io/status/1974647885456372118) I will keep this in mind once I return to the development in the near future

**Reply by Romasato** [reply]
Hi @lawand-dot-io,

Sorry to hear about that, some things can wait in life, until then wishing you a speedy recovery!

**Reply by lawand-dot-io** [reply]
@Romasato thanks :) it's nothing serious, but I will just slow down the app's development for the moment. Stay tuned, more features will eventually be added and I will keep your request for this one in mind :) cheers

**Comment by ulan-yisaev** [comment]
Hi @lawand-dot-io,
Thanks a lot for building Taskbar — it completely transformed my macOS experience after switching from Windows. I love being able to switch between windows like in the Windows taskbar.

Just adding my voice to this feature request: I’d really appreciate an option to place new windows next to the currently active one. I group related windows (e.g. PyCharm + Terminal), but when new ones always go to the far right, it breaks my layout and slows me down. This small UX tweak would make a big difference :)

Thanks again for the great tool!

**Reply by lawand-dot-io** [reply]
Thanks for the feedback, I'm happy you find Taskbar useful :)

**Comment by marcellus-trixus** [comment]
Also another useful setting for this feature could be 'Group by App', meaning if I open a new task (window), it pops up right next the last task of the same app.
